### DISCUSSION TITLE
ffmpeg_set_codec_preferred: Codec mkv not found

### DISCUSSION THREAD

**Discussion by GreatBigWhiteWorld** [Motion-Project/motion#1778] [Q&A]
I've tried 'mkv, mp4, mpeg4, mspeg4, h264, mjpeg' so far, but I get the same error "Codec **** not found".
OS is opensuse Leap. Motion version 4.6.0
I put the same camera on another machine running opensuse TW and it works fine.
I've also installed ffmpeg, ffmpeg-4, ffmpeg-7 too.

**Comment by tosiara** [comment]
Could you send the full log with -d 9 ?

**Reply by GreatBigWhiteWorld** [reply]
[Here](https://gist.github.com/GreatBigWhiteWorld/e1703d46adbd466d4acca5c6df548ba1) it is.

**Reply by tosiara** [reply]
I cannot access the log anymore

**Comment by tosiara** [comment]
Please run the following command and paste here the full output:
```
dd if=/dev/urandom bs=1 count=38400 | ffmpeg -y -f rawvideo -pix_fmt yuv420p -video_size 160x160 -framerate 1 -i pipe:0 -vcodec mpeg4 test.avi
```

**Reply by GreatBigWhiteWorld** [reply]
> Please run the following command and paste here the full output:
> 
> ```
> dd if=/dev/urandom bs=1 count=38400 | ffmpeg -y -f rawvideo -pix_fmt yuv420p -video_size 160x160 -framerate 1 -i pipe:0 -vcodec mpeg4 test.avi
> ```
Sorry for late reply, [here](https://gist.github.com/GreatBigWhiteWorld/49bcb7b1833ed77cf2419522d4c9626e)'s the output.

**Reply by tosiara** [reply]
I cannot access the motion log anymore in the previous comment, and so cannot validate which ffmpeg libs your motion was built with
